### ISSUE TITLE
[Dropdown] Followup tasks

### ISSUE THREAD

**Issue by brentertz** [mineral-ui/mineral-ui#366]
As a Mineral UI developer, I want to complete Dropdown component related tasks that were not accomplished during the initial development

* Dropdown matches latest spec
    * Notably, top/bottom space in the menu should be removed
* A11y
    * arrow up/down does not work in voiceover mode when using keyboard to open button (when trigger is not focused)
        * Possible solution:
            * If focused trigger onOpen, aria nav would probably work
            * Probably bad for when component is controlled, but might be okay
* restoreFocus paradigm needs investigation
    * Possible solution: skip restoreFocus if click outside component
    * To illustrate, put a Popover and a Dropdown on the same page, open one, then click the trigger of the other. Notice that the one you first opened steals focus from the newly-opened one.
    * Also affects Popover

* Cross-browser issues
    * Safari/Firefox does not focus when click trigger with mouse, thus arrowUp/Down does not work
        * Possible solution: 
            * focus trigger on open
            * might also fix similar a11y issue
        * This could also be indicative that the user is using a mouse and thus it doesn't need fixed
    * Edge (maybe IE)
        * Menu open via up/down arrow does not function

* Should disabled Dropdown/Popover automatically pass disabled prop to trigger?

* Tests

**Comment by brentertz** [comment]
Dropdown a11y announcement improvements has been split off into another story.  #402
